### DISCUSSION TITLE
Is it possible to somehow only print xyz on first navigation into folder

### DISCUSSION THREAD

**Discussion by einarpersson** [starship/starship#4965] [Q&A]
Hi! New user here. 👋 

When I begin a work session by navigating into a project folder, or just open the terminal somewhere, I want a _high degree of context_ (such as "what node version is active", "what branches are available", "..."). But I don't need all that context every time I enter something into the terminal. Is it possible - would it be possible - to somehow present a different prompt if it is "first navigation" / "new terminal session" vs "just staying in same folder"?

I understand if this is behaviour is not within the scope of this project, but I just wanted to share / get thoughts on this. It feels pretty natural to want to have different amount of context when you begin a work session vs every time you enter something into the terminal. (and I understand that this is the whole point of a prompt and if you want something to just happen on startup or cd you could write a custom script - but then again it would be nice to reuse the starship formatting etc)

**Edit:**
I realize that an option to this behaviour would be to make it so that the Prompt is "sticky" somewhere and updated rather than printed on every new line. Don't know if this is possible either, though. I see something called transient prompt in the docs, but only re powershell, cmd and fish (I use bash currently, might change to zsh)

**Comment by wellkilo** [comment]
Starship itself renders the prompt each time the shell asks for one; it does not maintain a portable “first prompt after entering this directory” state. There are two workable patterns, depending on the behavior you actually want.

### 1. Keep the first/full prompt, collapse old prompts

This is the transient-prompt model. For Bash, Starship delegates it to `ble.sh` rather than implementing Bash prompt rewriting itself:

```bash
# ~/.bashrc — ble.sh 0.4+
bleopt prompt_ps1_transient=always:trim
bleopt prompt_ps1_final='$(starship module character)'
```

Your current prompt stays rich; after you execute a command, the previous prompt is redrawn as the compact final prompt. This addresses terminal clutter without losing context when you need it. Current Starship docs: [Transient Prompt in Bash](https://github.com/starship/starship/blob/80d9cae24f155348f158672a36c4c9d908e0a6eb/docs/advanced-config/README.md#transientprompt-and-transientrightprompt-in-bash).

### 2. Truly show a module only once per directory

That requires shell state. A custom Starship module can use a shell-maintained variable/marker, but Starship's subprocess cannot persist an environment change back into its parent shell. The reliable implementation is therefore a `PROMPT_COMMAND`/directory-change hook that updates state, then a custom module whose `when` command reads that state.

For example, in Bash you can track the last directory in a cache file and make `custom.session_context` print only when `$PWD` changes. Be aware that one global cache file is shared across terminals; include `$$` or a terminal/session ID in the filename if sessions must be independent.

For the original goal, transient prompt is simpler and already supported. “Sticky context elsewhere on screen” is a terminal multiplexer/status-bar feature rather than a normal shell prompt feature.

If the transient-prompt setup gives you the desired full-current/compact-history behavior, please mark this as accepted.
